service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -500,14 +499,10 @@ public ResponseEntity<ApiCreateCloudContextResult> createCloudContext(
         workspaceService.validateMcWorkspaceAndAction(userRequest, uuid, SamWorkspaceAction.WRITE);
 
     if (body.getCloudPlatform() == ApiCloudPlatform.AZURE) {
-      ApiAzureContext azureContext =
-          Optional.ofNullable(body.getAzureContext())
-              .orElseThrow(
-                  () ->
-                      new CloudContextRequiredException(
-                          "AzureContext is required when creating an azure cloud context for a workspace"));
+      AzureCloudContext azureContext =
+          Optional.ofNullable(body.getAzureContext()).map(AzureCloudContext::fromApi).orElse(null);
```

Comment:
Where is the null handled now?

Reply:
The `CreateAzureContextFlight` that `workspaceService.createAzureCloudContext` spins up can handle a null cloud context IFF the `bpm-azure-enabled` feature flag is turned on. I'm going to add a check here so that we throw if `bpm-azure-enabled` is false and the cloud context is null.

Reply:
Sounds good, but how does having bpm allow the cloud context to be null? Is it because it gets the cloud context from the BPM record?

Reply:
Yeah exactly, WSM fetches the cloud context coordinates from the billing profile record referenced by the workspace's spend profile id